plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/StyleCode/hooks.ts

```diff
@@ -0,0 +1,53 @@
+import { postMsg } from "@web/extensions/sidebar/utils";
+import JSON5 from "json5";
+import { useCallback, useEffect, useState } from "react";
+
+import { BaseFieldProps } from "../../types";
+
+export default ({
+  value,
+  dataID,
+  onUpdate,
+}: Pick<BaseFieldProps<"styleCode">, "value" | "dataID" | "onUpdate">) => {
+  const [code, editCode] = useState(value.src);
+
+  const updateStyle = useCallback(
+    (styleStr: string) => {
+      if (styleStr) {
+        try {
+          const styleObject = JSON5.parse(styleStr);
+          postMsg({ action: "styleCodeUpdate", payload: { dataID, styleObject } });
+          // eslint-disable-next-line no-empty
+        } catch (error) {}
```

Comment:
This try/catch is not in an async function. Don't try/catches need to be in an async func?

Reply:
Does try/catch need to be in an async func? 😱

Reply:
Huh....apparently not. Why did I think they needed to be in async funcs.... stupid lol
Anyways, no they don't I guess :P